functions.sql

```diff
@@ -0,0 +1,86 @@
+/* Additional database functions for openstreetmap-carto */
+
+/* Access functions below adapted from https://github.com/imagico/osm-carto-alternative-colors/tree/591c861112b4e5d44badd108f4cd1409146bca0b/sql/roads.sql */
+
+/* Simplified 'yes', 'destination', 'no', 'unknown', NULL scale for access restriction 
+  'no' is returned if the rendering for highway category does not support 'restricted'.
+  NULL is functionally equivalent to 'yes', but indicates the absence of a restriction 
+  rather than a positive access = yes. 'unknown' corresponds to an uninterpretable 
+  access restriction e.g. access=unknown or motorcar=occasionally  */
+CREATE OR REPLACE FUNCTION carto_int_access(int_highway text, accesstag text)
+	RETURNS text
+	LANGUAGE SQL
+	IMMUTABLE PARALLEL SAFE
+AS $$
+SELECT
+	CASE
+	WHEN accesstag IN ('yes', 'designated', 'permissive') THEN 'yes'
+	WHEN accesstag IN ('destination',  'delivery', 'customers') THEN
+		CASE WHEN int_highway IN ('road', 'pedestrian') THEN 'restricted' ELSE 'no' END
```

Comment:
I had overseen in my previous review that this right now changes the interpretation of `access=destination` for the paths/tracks with the two class access rendering from `yes` to `no`.  Since this is not the intention of this PR or of #214 we should change that back - simply by changing the 'no' to 'yes' here.

Reply:
Agreed. Commit made.